packages/loader/container-loader/src/loader.ts

```diff
@@ -276,10 +276,22 @@ export class Loader extends EventEmitter implements ILoader {
     public async createDetachedContainer(codeDetails: IFluidCodeDetails): Promise<Container> {
         debug(`Container creating in detached state: ${performance.now()} `);
 
-        return Container.createDetached(
+        const container = await Container.createDetached(
             this,
             codeDetails,
-            );
+        );
+
+        if (this.cachingEnabled) {
+            container.once("attached", () => {
+                ensureFluidResolvedUrl(container.resolvedUrl);
+                const parsedUrl = parseUrl(container.resolvedUrl.url);
+                if (parsedUrl !== undefined) {
+                    this.containers.set(parsedUrl.id, Promise.resolve(container));
```

Comment:
Looks like this could cause false misses in the cache when requesting for the same container later if a version is set in the request header.  That path uses getKeyForContainerCache which will append that version value.  Though that's getting removed as part of the headers cleanup so maybe nbd

Reply:
I spent a little time looking at this - we don't currently have any examples of setting version on the `attach` request (and semantically I don't know what it would mean really, since the `attach` call implies there is no version to load from yet anyway).  So `getKeyForContainerCache` is always just the ID in the `attach` flow currently.

Separately, I don't really understand why we'd want distinct entries in the cache based on the specified version.  If I understand the scenario correctly, it seems to me that specifying the version should probably bypass reading from the cache and only use the ID when writing to the cache (omitting the version).  Or maybe even bypass writing to the cache entirely, treating specified-version loading as a "one-off" load.